src/server/config/schema.js

```diff
@@ -138,7 +138,7 @@ export default () => Joi.object({
       }),
       keyPassphrase: Joi.string(),
       certificateAuthorities: Joi.array().single().items(Joi.string()).default([]),
-      supportedProtocols: Joi.array().items(Joi.string().valid('TLSv1', 'TLSv1.1', 'TLSv1.2')),
+      supportedProtocols: Joi.array().items(Joi.string().valid('TLSv1', 'TLSv1.1', 'TLSv1.2')).default(['TLSv1.1', 'TLSv1.2']),
```

Comment:
__issue:__ okay, I see two issues here:

The first is that SSL is completely handled in the new platform now and validated by `src/core/server/http/ssl_config.ts`, so default value should be set there. If we set it here it won't be used anywhere. Sorry for the confusion, that's totally my fault in not making this critical thing crystal clear in the code. So what I'd propose is to change `ssl` schema in the legacy platform to something like this (/cc @epixa):
```js
ssl: Joi.any().description('the key is handled in the new platform ONLY')
```
We could also remove this key from the config that's being passed to the legacy world by `LegacyService` that would allow us to completely remove this old schema, **but** we currently rely on SSL related deprecation warnings issued by the legacy platform and it needs to have access to this config key.

For the new platform schema would look like:
```ts
supportedProtocols: schema.arrayOf(
  schema.oneOf([schema.literal('TLSv1'), schema.literal('TLSv1.1'), schema.literal('TLSv1.2')]),
  { defaultValue: ['TLSv1.1', 'TLSv1.2'] }
),
```
And `SslConfig.supportedProtocols: string[] | undefined` can become `SslConfig.supportedProtocols: string[]`

The second issue is that CI is green, while it should be red, we probably need to write some test to test such important defaults in this PR in a follow-up :)

Reply:
Btw, I see we have legacy tests for SSL config, we can just move them to the new platform

Reply:
I went ahead and moved the logic in the new platform to use the suggested schema syntax.

How would you feel about decoupling the removal of the legacy ssl settings in another PR?

Reply:
> How would you feel about decoupling the removal of the legacy ssl settings in another PR?\
Sounds good to me.